### PR TITLE
Add unit tests for Helix SDK XHarness tasks

### LINKED ISSUE #7060: CreateXHarnessAndroidWorkItems fails if previous .zip file exists
- [ ] This issue is blocking 
- [ ] This issue is causing unreasonable pain

I'm experimenting with xharness etc for use in perf testing. While kicking the tires I hit a crash that only reproduces when the `zip` archive created from a previous run still exists:
```❯ dotnet msbuild test.proj
Microsoft (R) Build Engine version 16.8.3+39993bd9d for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Tool 'microsoft.dotnet.xharness.cli' is already installed.
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018: The "CreateXHarnessAndroidWorkItems" task failed unexpectedly. [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018: System.IO.InvalidDataException: A local file header is corrupt. [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at System.IO.Compression.ZipArchive..ctor(Stream stream, ZipArchiveMode mode, Boolean leaveOpen, Encoding entryNameEncoding) [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at System.IO.Compression.ZipArchive..ctor(Stream stream, ZipArchiveMode mode) [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at Microsoft.DotNet.Helix.Sdk.XHarnessTaskBase.AddToPayloadArchive(String payloadArchivePath, String targetFilename, Stream content) in /_/src/Microsoft.DotNet.Helix/Sdk/XharnessTaskBase.cs:line 107 [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at Microsoft.DotNet.Helix.Sdk.XHarnessTaskBase.AddResourceFileToPayload(String payloadArchivePath, String resourceFileName, String targetFileName) in /_/src/Microsoft.DotNet.Helix/Sdk/XharnessTaskBase.cs:line 101 [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at Microsoft.DotNet.Helix.Sdk.CreateXHarnessAndroidWorkItems.CreateZipArchiveOfPackageAsync(String fileToZip) in /_/src/Microsoft.DotNet.Helix/Sdk/CreateXHarnessAndroidWorkItems.cs:line 92 [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at Microsoft.DotNet.Helix.Sdk.CreateXHarnessAndroidWorkItems.PrepareWorkItem(ITaskItem appPackage) in /_/src/Microsoft.DotNet.Helix/Sdk/CreateXHarnessAndroidWorkItems.cs:line 68 [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at Microsoft.DotNet.Helix.Sdk.CreateXHarnessAndroidWorkItems.ExecuteAsync() in /_/src/Microsoft.DotNet.Helix/Sdk/CreateXHarnessAndroidWorkItems.cs:line 42 [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at Microsoft.DotNet.Helix.Sdk.CreateXHarnessAndroidWorkItems.Execute() in /_/src/Microsoft.DotNet.Helix/Sdk/CreateXHarnessAndroidWorkItems.cs:line 32 [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\source\performance\test.proj]
C:\Users\billwert\.nuget\packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21125.5\tools\xharness-runner\XHarnessRunner.targets(96,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\source\performance\test.proj]
```

Repro:
1) Run a test (maybe the test needs to fail to upload as mine currently are)
2) Run the test again

If I delete the zip file it gets past this.

### PR DESCRIPTION
**Pull request by premun**
**Motivation**
XHarness tasks in Helix SDK are getting more and more complex and we don't have them tested. In order to prepare for even more new functionality (https://github.com/dotnet/xharness/issues/590), we need to add unit tests.

**Changes**
- No functional changes on the outside
- Refactoring of classes to be testable
- Tests for `CreateXHarnessAndroidWorkItemsTests` and `CreateXHarnessAppleWorkItemsTests`
- New class `ProvisioningProfileProvider` + tests
- Fixes #7060

### CODE REVIEW

File: src/Common/Microsoft.Arcade.Common/IZipArchiveManager.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        /// <param name="resourceName">Name of the embedded resource</param>
+        /// <param name="targetFileName">New name of the file in the archive</param>
+        Task AddResourceFileToArchive<TAssembly>(string archivePath, string resourceName, string targetFileName = null);
+
+        /// <summary>
+        /// Compresses a directory into an archive on a given path.
+        /// </summary>
+        /// <param name="directoryPath">The directory to archive</param>
+        /// <param name="archivePath">Path where to create the archive</param>
+        /// <param name="includeBaseDirectory">Should the top level entry of the arvhive be the directory?</param>
```
Comment (MattGal) [inline src/Common/Microsoft.Arcade.Common/IZipArchiveManager.cs:21]: Nits:  'arvhive', also not usually expressed as a question.  Maybe more like "When true, includes top-level directory in the archive"?

File: src/Common/Microsoft.Arcade.Common/ZipArchiveManager.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+using System.IO;
```
Comment (MattGal) [inline src/Common/Microsoft.Arcade.Common/ZipArchiveManager.cs:1]: License header?

File: src/Common/Microsoft.Arcade.Common/IZipArchiveManager.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+using System.Threading.Tasks;
```
Comment (MattGal) [inline src/Common/Microsoft.Arcade.Common/IZipArchiveManager.cs:1]: License header? (going to stop commenting this, consider fixing them...)

**Review by MattGal — APPROVED** [review]
Looks good (it's huge so I skimmed most of it).  Lots of missing license headers.

File: src/Common/Microsoft.Arcade.Common/FileSystem.cs [hunk 4]
```diff
@@ -3,30 +3,38 @@
...
+#nullable enable
 namespace Microsoft.Arcade.Common
 {
     public class FileSystem : IFileSystem
     {
-        public void CreateDirectory(string path)
-        {
-            Directory.CreateDirectory(path);
-        }
+        public void CreateDirectory(string path) => Directory.CreateDirectory(path);
```
Comment (missymessa) [inline src/Common/Microsoft.Arcade.Common/FileSystem.cs:11]: Oh these are nice refactors. A lot more modern :)

**Review by missymessa — APPROVED** [review]
LGTM shipit

**Review by greenEkatherine — APPROVED** [review]

**Pull request merged**